src/SignalR/server/Core/src/HubOptions.cs

```diff
@@ -70,4 +73,13 @@ public int MaximumParallelInvocationsPerClient
             _maximumParallelInvocationsPerClient = value;
         }
     }
+
+    /// <summary>
+    /// When enabled, will use <see cref="IServiceProviderIsService"/> to guess if a Hub method parameter can be injected from the DI container.
+    /// Parameters can be explicitly marked with an attribute that implements <see cref="IFromServiceMetadata"/> with or without this option enabled.
+    /// </summary>
+    /// <remarks>
+    /// Disabled by default.
+    /// </remarks>
+    public bool EnableInferredFromServiceParameters { get; set; }
```

Comment:
I think we should infer service parameters by default. I'd be tempted to remove this option altogether. How many service types are also serializable? I see wanting an option to opt-out though since there isn't another escape hatch I can see if you happen to have an argument type registered as a service.

Reply:
> I see wanting an option to opt-out though since there isn't another escape hatch I can see if you happen to have an argument type registered as a service.

This is my main concern, which is why I switched the whole PR to using `IFromServiceMetadata` and then having an opt-in for inferring.

Reply:
Lets leave the opt-out then.

Reply:
I'm going to leave as opt-in for now, we can discuss in chat and/or API review